src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,380 @@
+      val newSnapshot = RecentSnapshot("aaaa", 2L, Map.empty)
 
-      val persist = redownloadService.persistOwnSnapshot(2L, "aaaa")
-      val check = redownloadService.getOwnSnapshot(2L)
+      val persist = redownloadService.persistLocalSnapshot(newSnapshot)
+      val check = redownloadService.getLocalSnapshotAtHeight(2L)
 
-      (persist >> check).unsafeRunSync shouldBe "aaaa".some
+      (persist >> check).unsafeRunSync shouldBe RecentSnapshot("aaaa", 2L, Map.empty).some
     }
 
     "should return None if snapshot at given height does not exist" in {
-      val redownloadService = RedownloadService[IO]()
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
 
-      val check = redownloadService.getOwnSnapshot(2L)
+      val check = redownloadService.getLocalSnapshotAtHeight(2L)
 
       check.unsafeRunSync shouldBe none[String]
     }
   }
 
+  "fetchAndSetPeersProposals" - {
+    "should fetch peer proposals and modify proposal map" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (_, peerApi) =>
+          peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF serializedResponse
+      }
+      val fetch = redownloadService.fetchAndSetPeerProposals()
+      val check = redownloadService.proposedSnapshots.get
+      (fetch >> check).unsafeRunSync().keySet shouldBe facilitators.keySet
+    }
+
+    "should not fail if at least one peer did not respond" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.tail.foreach {
+        case (_, peerApi) =>
+          peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF serializedResponse
+      }
+      facilitators.head._2.client.getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturn IO
+        .raiseError(new TimeoutException("Testing timeout, case just ignore this message."))
+      val fetch = redownloadService.fetchAndSetPeerProposals()
+      val check = redownloadService.proposedSnapshots.get
+      (fetch >> check).unsafeRunSync().keySet shouldBe facilitators.tail.keySet
+    }
+  }
+
+  "updatePeerProposals" - {
+    "should update peersProposals" in {
+      val redownloadService = RedownloadService[IO](cluster, healthChecker)
+      cluster.readyPeers shouldReturn IO.pure[Map[Id, PeerData]](facilitators)
+      facilitators.foreach {
+        case (_, peerApi) =>
+          peerApi.client
+            .getNonBlockingFLogged[IO, Array[Array[Byte]]](*, *, *, *)(*)(*, *, *) shouldReturnF serializedResponse
+      }
+      val fetch = redownloadService.fetchAndSetPeerProposals()
```

Comment:
Why `updatePeerProposals` needs `fetchAndSetPeerProposals`?

You shouldn’t test private methods directly, but only their effects on the public methods that call them. Unit tests are clients of the object under test, much like the other classes in the code that are dependent on the object. The test should only be accessing the class’ public interface. If an object is hard to test via its public interface, it is going to be hard to use in the production code. This is one of many examples where a unit test is exposing potential design issues.